public/visualization.component.ts

```diff
@@ -8,189 +14,330 @@ import { find, values } from "lodash";
+        zEntries.sort(function(a, b) {
+            return b.r - a.r;
+        });
+
+        for (let i in zEntries) {
+            let zEntry = zEntries[i];
             // todo: hover boundary of 2px
-            const circle = d3.select('svg').append('circle')
-                .attr('cx', x)
-                .attr('cy', y)
-                .attr('r', sizeByLength ? Math.max(2, entry.biographyLength * .02): sizeByTravelTime ? Math.max(2, entry.travelTime * .02): 0)
-                .attr('fill', 'grey')
+            d3.select('svg').append('circle')
+                .attr('cx', zEntry.cx)
+                .attr('cy', zEntry.cy)
+                .attr('r', zEntry.r)
+                .attr('fill', zEntry.fill)
+                .style("opacity", 0.75)
                 // we define "mouseover" handler, here we change tooltip
                 // visibility to "visible" and add appropriate test
 
                 .on("mouseover", function (d) {
                     div.transition()
-                        .duration(200)
-                        .style("opacity", .9);
-                    div.html(entry.fullName)
+                        .style("opacity", .9)
+                    div.text(zEntry.fullName)
                         .style("left", (d3.event.pageX) + "px")
-                        .style("top", (d3.event.pageY - 28) + "px");
-                })
+                        .style("top", (d3.event.pageY - 28) + "px")
+                        .style("opacity", .9)
+                    }
+                )
                 .on("mouseout", function (d) {
                     div.transition()
-                        .duration(500)
                         .style("opacity", 0);
-                })
+                    }
+                )
                 .on("click", function() {
                     div.style("opacity", 0);
-                    window.location.hash = `/entries/${entry.index}`;
-                });
+                    var hash = `/#/entries/${zEntry.index}`;
+                    window.open(hash);
+                    }
+                );
         }
-        if (!sizeByLength && !sizeByTravelTime) {
-            d3.select('svg').selectAll('circle').transition().attr('r', 3).duration(1000);   
-        }
-        return { x, y };
+
+        return y; // The placement of the lowest dot is returned. This is used to position the next group.
     }
 
-    private async groupByType(queryOptions) {
-        this.loading = true;
-        let entriesList;
+    /*
+     * Using queries from getGroups, all entries are mapped to the appropriate group and returned.
+     */
+    private async groupByType(allGroups) {
+        let entryGroups;
         try {
-            entriesList = await Promise.all(queryOptions.map(queryOption =>
-                this.http.post('/api/entries/search', { query: queryOption.query }).toPromise()
+            entryGroups = await Promise.all(allGroups.map(group =>
```

Comment:
Oh yeah, you can just replace this with 
```suggestion
            return await Promise.all(allGroups.map(group =>
```

and remove the `return entryGroups;` after the catch block.